website/src/APITesting/SignupAPi.test.js

```diff
@@ -0,0 +1,27 @@
+const reqData = {
+  email: 'Student6@gmail.com',
+  name: 'API Test',
+  password: '1234567890',
+  type: 'student',
+};
+
+function SignUpApi(input) {
+  const fetchData = fetch('https://us-central1-rails-students.cloudfunctions.net/signup', {
+    method: 'POST',
+    headers: {
+      Accept: 'application/json',
+      'Content-type': 'application/json',
+    },
+    body: JSON.stringify(input),
+  }).then(result => result.status);
+  return fetchData;
+}
+
+/* To check if a existed userdata on signup page should return 406 */
+describe('SignUp APi should work', () => {
+  test('The Signup API should return 406 when the user is signuped', async () => { // requir async, since javascipt will not wait for api completed
+    // await mean to wait which process to be done, await only can be use in async function
+    const data = await SignUpApi(reqData);
+    expect(data).toBe(406); // check th status
```

Comment:
These tests will result in users being signed up every single time we run this test and this test will only pass the first time you run it because the second time there will already be an existing user with the same information

Reply:
Ye, therefore, I change the meaning of SIGNUPAPI. This test is not testing the if the user is being signup successfully. It is tests the signupapi should returns 406 when a user signup his account when his account is in the database

Reply:
Oh so its checking that the signup failed.

Reply:
Thats not really a good metric to test because it could fail for any number of reasons.

Reply:
Maybe find out if there's a way to delete users from the database so testing creating a user and then deleting them. @ayushyamitabh  do you think this is possible

Reply:
No, can't read and write to database from client side, unsafe

Reply:
I mean what if there was an api route we could access to delete users. Like a dev only route.

Reply:
@khelif96  you mean like https://console.firebase.google.com lol?

Reply:
No i mean a route he can call in his test after verifying the user has been created to delete that user